modules/renter/contractor/uptime.go

```diff
@@ -0,0 +1,51 @@
+package contractor
+
+import (
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+)
+
+// uptimeInterval specifies how frequently hosts are checked.
+var uptimeInterval = func() time.Duration {
+	switch build.Release {
+	case "dev":
+		return 1 * time.Minute
+	case "standard":
+		return 30 * time.Minute
+	case "testing":
+		return 100 * time.Millisecond
+	}
+	panic("undefined uptimeInterval")
+}()
+
+// threadedMonitorUptime regularly checks host uptime, and deletes contracts
+// whose hosts fall below a minimum uptime threshold.
+func (c *Contractor) threadedMonitorUptime() {
```

Comment:
Is deleting the best course of action? I feel like it might be better just to mark the contract as 'unusable'. Then the signal is there for the renter that new contracts need to be created, but you are still able to talk to the host in the future if it comes back.